modules/porous_flow/src/materials/PorousFlowCapillaryPressureLinear.C

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowCapillaryPressureLinear.h"
+
+template<>
+InputParameters validParams<PorousFlowCapillaryPressureLinear>()
+{
+  InputParameters params = validParams<PorousFlowCapillaryPressureBase>();
+  params.addRequiredParam<Real>("pc_max",  "Maximum capillary pressure (Pa)");
+  params.addClassDescription("This Material provides a linear capillary pressure");
+  return params;
+}
+
+PorousFlowCapillaryPressureLinear::PorousFlowCapillaryPressureLinear(const InputParameters & parameters) :
+  PorousFlowCapillaryPressureBase(parameters),
+  _pc_max(getParam<Real>("pc_max"))
+{
+}
+
+void
+PorousFlowCapillaryPressureLinear::computeQpProperties()
+{
+  /// Capillary pressure and derivatives wrt phase saturation at the nodes
```

Comment:
In the Dictator (which every porous-flow simulation needs), could i do the loop over the mesh?  Not that i actually know how to do that, but it would be a good place.

for (elem = 0; elem < number_elements; ++elem)
  if (elem.numberQPs() != elem.numberNodes())
    CRASH